src/smart-components/common/edit-approval-workflow.tsx

```diff
@@ -54,13 +56,22 @@ const EditApprovalWorkflow: React.ComponentType<EditApprovalWorkflowProps> = ({
 
   const onSubmit = (toLink: string[], toUnlink: string[]) => {
     close();
-    dispatch(
-      updateWorkflows(toLink, toUnlink, {
-        object_type: objectType,
-        app_name: APP_NAME[objectType],
-        object_id: query[querySelector]
-      })
+    const promise = new Promise((resolve) =>
+      resolve(
+        dispatch(
```

Comment:
You can call `.then` on the dispatch function. The redux promise middleware returns a promise. Typescript is just having a hard time with inferring redux action creators modified by middlewares.